src/concepts/essentials/workflows/sdk.md

```diff
@@ -0,0 +1,1177 @@
+
+#### `reporter: string;`
+
+- The `reporter` property is of type `string` and identifies what discovered the Finding (_for example, the Reporter value will be a Workflow_).
+
+#### `request: Request;`
+
+- `request` stores the [Request](#request) object.
+
+## FindingsSDK
+
+- The `FindingsSDK` type interface provides a method to create a new Finding.
+
+```
+export declare type FindingsSDK = {
+  create(spec: FindingSpec): Promise<Finding>;
+};
+```
+
+#### `create(spec: FindingSpec): Promise<Finding>;`
+
+- This method will use the `spec` parameter with a type of [FindingSpec](#findingspec). The return type is a Promise (since the function is asynchronous) that resolves as [Finding](#finding). If the request cannot be saved - an error will be logged automatically.
+
+::: tip
+Example:
+
+```js
+sdk.findings.create({
+  title: "Title",
+  description: "Description",
+  reporter: "Reporter",
+  request,
+});
+```
+
+:::
+
+## Types
+
+#### `HttpInput`
+
+```
+export declare type HttpInput = {
+  request: Request | undefined;
+  response: Response | undefined;
+};
+```
+
+#### `BytesInput`
+
+```
+export declare type BytesInput = Array<number>;
+```
+
+#### `ID`
+
+```
+export declare type ID = string;
+```
+
+#### `Data`
+
+```
+export declare type Data = Bytes;
+```
+
+#### `Decision`
+
+```
+export declare type Decision = boolean;
+```
+
+#### `Bytes`
+
+```
+export declare type Bytes = string | Array<number> | Uint8Array;
+```
+
+#### `MaybePromise<T>`
+
+- Allows for the handling of both synchronous and asynchronous values.
+
+```
+export declare type MaybePromise<T> = T | Promise<T>;
+```
+
+## SDK
+
+The `SDK` object is an interface that provides access to various services and functionalities.
+
+```
+export declare type SDK = {
+  console: Console;
+  findings: FindingsSDK;
+  requests: RequestsSDK;
+  asString(array: Bytes): string;
+};
+```
+
+#### `console: Console;`
+
+- The [Console](#console) object for logging.
+
+::: tip
+
+Examples:
+
+```js
+sdk.console.debug();
+sdk.console.log();
+sdk.console.warn();
+sdk.console.error();
+```
+
+:::
+
+#### `findings: FindingsSDK;`
+
+- The [FindingsSDK](#findingssdk) for interacting with Findings.
+
+::: tip
+Example:
+
+```js
+sdk.findings.create({
+  title: "Title",
+  description: "Description",
+  reporter: "Reporter",
+  request,
+});
+```
+
+:::
+
+#### `requests: ReequestsSDK;`
+
+- The [RequestsSDK](#requestssdk) for interacting with requests.
+
+::: tip
+Example:
+
+```js
+const spec = new RequestSpec(`"https://example.com"`);
+sdk.requests.send(spec)
+  .then((res) => {
+    sdk.console.log(res.request.getId());
+    sdk.console.log(res.response.getCode());
+  })
+  .catch((err) => {
+    sdk.console.error(err);
+  });
+```
+
+:::
+
+#### `asString(array: Bytes): string;`
+
+- A helper function that converts Bytes (`string/Array<number>/Uint8Array`) to a string, replacing unprintable characters with `�`.
```

Comment:
Thats the only tip I would keep since its a direct method call

Reply:
I do feel like the send() method example is beneficial. At least for newcomers. Which I feel like it's mostly going to be newcomers reading this page in general lol.

Reply:
Thats fine